commit 09c8e49acb5c00a6c9893deeca4c3f41df6dfa8b
Author: Jakub Jankowski <jakub.jankowski@superhost.pl>
Date:   Mon Jun 6 13:08:56 2011 +0200

    Fix double ntohs() in SMTP checker
    
    Double ntohs() call in smtp_dump_host() caused printing wrong value for
    RS checked port in --dump-conf output (ie. "port = 6400" instead of 25).
    
    Signed-off-by: <jakub.jankowski@superhost.pl>

diff --git a/keepalived/check/check_smtp.c b/keepalived/check/check_smtp.c
index d8eb954..d8bd02e 100644
--- a/keepalived/check/check_smtp.c
+++ b/keepalived/check/check_smtp.c
@@ -70,7 +70,7 @@ smtp_dump_host(void *data)
 {
         smtp_host_t *smtp_host = data;
         log_message(LOG_INFO, "   Checked ip = %s", inet_sockaddrtos(&smtp_host->dst));
-        log_message(LOG_INFO, "           port = %d", ntohs(ntohs(inet_sockaddrport(&smtp_host->dst))));
+        log_message(LOG_INFO, "           port = %d", ntohs(inet_sockaddrport(&smtp_host->dst)));
 	if (smtp_host->bindto.ss_family)
         	log_message(LOG_INFO, "           bindto = %s", inet_sockaddrtos(&smtp_host->bindto));
 }
